machine-learning/02-Data Preparation and Feature Engineering.ipynb

```diff
@@ -1,45 +1,405 @@
+        "        Rate_code = 1,\n",
+        "        Passenger_count = 1,\n",
+        "        Trip_time_in_secs = 474,\n",
+        "        Trip_distance = 1.5f,\n",
+        "        Payment_type = \"CRD\",\n",
+        "        Fare_amount = 8, \n",
+        "    }\n",
+        "};"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "dotnet_interactive": {
+          "language": "csharp"
+        },
+        "vscode": {
+          "languageId": "dotnet-interactive.csharp"
+        }
+      },
+      "outputs": [],
+      "source": [
+        "// Create MLContext\n",
+        "MLContext mlContext = new MLContext();\n",
+        "\n",
+        "//Load Data\n",
+        "IDataView data = mlContext.Data.LoadFromEnumerable<ModelInput>(inMemoryCollection);\n",
+        "\n",
+        "data.Preview(1);"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### What's the difference between a DataFrame and IDataView?\n",
+        "\n",
+        "You may have heard of the [DataFrame](https://docs.microsoft.com/en-us/dotnet/api/microsoft.data.analysis.dataframe?view=ml-dotnet-preview) type. It is another tool to load, view and manipulate data that is common to Notebooks. It implements an IDataView, so it can easily be passed to ML.NET.\n",
+        "\n",
+        "DataFrame and IDataView are very similar in the sense that they both are ways of representing data in a tabular format and applying transformations for it. Some key differences:\n",
+        "\n",
+        "- DataFrame only supports loading delimited files.\n",
+        "- DataFrame runs on memory so you're limited to the amount of memory on your PC.\n",
+        "\n",
+        "The DataFrame is recommended when performing tasks like exploratory data anlysis on a sample of your data. Look at the reference notebook REF - Data Processing for an example of using Data Frames to manipulate a data file for training.\n",
+        "\n",
+        "IDataView is recommended for training on larger datasets, and what we will use here for training examples. "
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## Data Transformations\n",
+        "\n",
+        "ML.NET supports a variety of data transformations that will convert data into the required format and help you make corrections to your data. Some common operations are manipulating columns, normalizing values, replacing missing values, converting values, and more. \n",
+        "\n",
+        "Look for more information on [data transformations available](https://docs.microsoft.com/en-us/dotnet/machine-learning/resources/transforms). \n",
+        "\n",
+        "Below are a few common transformations. "
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "### One Hot Encoding \n",
```

Comment:
I would recommend adding a header above this describing the type of data you're working with. OneHotEncoding is often used with categorical data.